dev-portal/scripts/deploy-stack.js

```diff
@@ -1,48 +1,72 @@
-const samTemplate = deployerConfig.samTemplate || r(`../../cloudformation/template.yaml`)
-const packageConfig = deployerConfig.packageConfig || r(`../../cloudformation/packaged.yaml`)
-const customersTableName = deployerConfig.customersTableName || 'DevPortalCustomers'
+const samTemplate = getOptional('samTemplate', r(`../../cloudformation/template.yaml`))
+const packageConfig = getOptional('packageConfig', r(`../../cloudformation/packaged.yaml`))
+const customersTableName = getOptional('customersTableName', 'DevPortalCustomers')
 
 // optional inputs
-const cognitoDomainName = deployerConfig.cognitoDomainName || ''
-const staticAssetRebuildMode = deployerConfig.staticAssetRebuildMode || ''
+const staticAssetRebuildMode = getOptional('staticAssetRebuildMode', '')
 
 // AWS SAM CLI configuration
-const awsSamCliProfile = deployerConfig.awsSamCliProfile;
-const profileOption = awsSamCliProfile ? `--profile ${awsSamCliProfile}` : ''
-
-function main() {
-  Promise.resolve()
-      .then(() => execute(`sam package --template-file ${samTemplate} --output-template-file ${packageConfig} --s3-bucket ${buildAssetsBucket} ${profileOption}`, true))
-.then(() => execute(`sam deploy --template-file ${packageConfig} --stack-name ${stackName} --capabilities CAPABILITY_NAMED_IAM --parameter-overrides StaticAssetRebuildToken="${Date.now()}" StaticAssetRebuildMode="${staticAssetRebuildMode}" DevPortalSiteS3BucketName="${siteAssetsBucket}" ArtifactsS3BucketName="${apiAssetsBucket}" DevPortalCustomersTableName="${customersTableName}" CognitoDomainNameOrPrefix="${cognitoDomainName}" --s3-bucket ${buildAssetsBucket} ${profileOption}`, true))
-.then(() => writeConfig(true))
-.then(() => console.log('\n' + 'Process Complete! Run `npm run start` to launch run the dev portal locally.\n'.green()))
-.catch(err => {
-    console.log(("" + err).red())
-  })
+const awsSamCliProfile = getOptional('awsSamCliProfile')
+
+async function main() {
+  execute('sam', [
+    'package',
+    '--template-file', samTemplate,
+    '--output-template-file', packageConfig,
+    '--s3-bucket', buildAssetsBucket,
+    ...awsSamCliProfile ? ['--profile', awsSamCliProfile] : []
+  ])
+  execute('sam', [
+    'deploy',
+    '--template-file', packageConfig,
+    '--stack-name', stackName,
+    '--capabilities', 'CAPABILITY_NAMED_IAM',
+    '--parameter-overrides',
+    `StaticAssetRebuildToken=${Date.now()}`,
+    ...staticAssetRebuildMode ? [`StaticAssetRebuildMode=${staticAssetRebuildMode}`] : [],
+    `DevPortalSiteS3BucketName=${siteAssetsBucket}`,
+    `ArtifactsS3BucketName=${apiAssetsBucket}`,
+    `DevPortalCustomersTableName=${customersTableName}`,
+    `CognitoDomainNameOrPrefix=${cognitoDomainName}`,
+    '--s3-bucket', buildAssetsBucket,
+    ...awsSamCliProfile ? ['--profile', awsSamCliProfile] : []
+  ])
+  await writeConfig()
+  console.log()
+  console.log(green('Process Complete! Run `npm run start` to launch run the dev portal locally.'))
+  console.log()
```

Comment:
For spacing? Why not `console.log('\n', blah, '\n')`?

Reply:
That and to make it align more roughly with what's actually printed to console.